gui/src/app/requests/request-detail/request-detail.component.ts

```diff
@@ -66,20 +75,43 @@ export class RequestDetailComponent implements OnInit {
                 console.log(error);
             });
         });
-        //TODO add translation
         this.translate.get("REQUEST.REQUEST_ERROR").subscribe(value => this.noCommentErrorMessage = value);
+        this.translate.get("REQUEST.REJECTED").subscribe(value => this.successRejcetMessage = value);
+        this.translate.get("REQUEST.APPROVED").subscribe(value => this.successApproveMessage = value);
+        this.translate.get("REQUEST.SET_WFC").subscribe(value => this.successSetWFCMessage = value);
         this.isUserAdmin = AppComponent.isUserAdmin();
-        //this.configService.isUserAdmin().subscribe(bool => this.isUserAdmin = bool);
     }
 
     ngOnDestroy(): void {
         this.sub.unsubscribe();
     }
 
+    openApproveDialog(): void {
+      const dialogRef = this.dialog.open(RequestDetailDialogComponent, {
+        width: '200px',
+        data: {isApprove: true, parent: this}
+      });
+
+      dialogRef.afterClosed().subscribe(result => {
+        console.log('The dialog was closed');
+      });
+    }
+
+    openRejectDialog(): void {
+      const dialogRef = this.dialog.open(RequestDetailDialogComponent, {
+        width: '250px',
+        data: {isApprove: false, parent: this}
+      });
+
+      dialogRef.afterClosed().subscribe(result => {
+        console.log('The dialog was closed');
+      });
+    }
+
     reject() {
         if (this.isUserAdmin) {
```

Comment:
If a user is not an admin, you should not even display the button to reject a request. Thus this check is unnecessary.